### PR TITLE
M3-5603: Clean up GooglePay code

### PR DESCRIPTION
**Pull request by HanaXu**
## Description
Cleans up some Google Pay code

## How to test
Go to `account/billing` and ensure that the following still works:
- Adding GooglePay as a recurring payment method
- Making a one time payment with the GooglePay Button
- Error handling (toasts and notices)

### CODE REVIEW

File: packages/manager/src/features/Billing/GooglePayProvider.ts [hunk 1]
```diff
@@ -119,31 +131,48 @@ export const gPay = async (
...
     }
 
     const errorMsg = isOneTimePayment
       ? 'Unable to complete Google Pay payment'
       : 'Unable to add payment method';
-
-    reportException(error, {
-      message: errorMsg,
-    });
     // @TODO Consider checking if error is an APIError so we can provide a more descriptive error message.
```
Comment (bnussman) [inline packages/manager/src/features/Billing/GooglePayProvider.ts:176]: We may want to address this `TODO`. I looked at what `api-v4` may return and there are some error messages that may be valuable to the user.

Comment (HanaXu) [reply packages/manager/src/features/Billing/GooglePayProvider.ts:176]: What error messages may `api-v4` return?

Comment (bnussman) [reply packages/manager/src/features/Billing/GooglePayProvider.ts:176]: Looks like messages regarding invalid address information for a customer

`ZIP/Postal code contains invalid characters; only letters, numbers, spaces and hyphens are allowed. Please update your account and try again.`

`Please update your account and try again.`

**Review by codyfinn — APPROVED** [review]
Nice job cleaning up the code!

**Review by codyfinn — COMMENTED** [review]
Did not mean to approve yet

**Review by tiffwong — APPROVED** [review]
Everything works as expected, fantastic job! 💯

**Review by codyfinn — APPROVED** [review]

**Review by bnussman — APPROVED** [review]

**Pull request merged**